lib/Parse/ParseStmt.cpp

```diff
@@ -1274,8 +1319,22 @@ ParserResult<PoundAvailableInfo> Parser::parseStmtConditionPoundAvailable() {
                          diag::avail_query_expected_rparen, LParenLoc))
     Status.setIsParseError();
 
+  // Diagnose #available == false as being a wrong spelling of #unavailable.
+  if (!isUnavailability && 
+      Tok.isAnyOperator() && 
+      Tok.getText() == "==" &&
+      peekToken().is(tok::kw_false)) {
+    diagnose(Tok, diag::false_available_is_called_unavailable)
+      .highlight(SourceRange(PoundLoc, peekToken().getLoc()))
+      .fixItReplace(PoundLoc, getTokenText(tok::pound_unavailable))
+      .fixItRemove(SourceRange(Tok.getLoc(), peekToken().getLoc()));
+    consumeToken();
+    consumeToken();
+    Status.setIsParseError();
```

Comment:
Could we actually consider this as `#unavailable`? In other words, could we `isUnavailability = true` here?

Reply:
It depends. The thing is that we can only detect this after parsing the list, and the list itself is parsed differently depending on what the token is (ignoring lacking wildcard). It would work if peeked the entire expression before parsing it, but I don't know if that would be a good idea. What do you think?

Reply:
Thanks for clarifying. I don't think it's worth it.